core/py/src/braintrust_core/json.py

```diff
@@ -0,0 +1,60 @@
+import dataclasses
+import json
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel
+
+
+class JSONSerializationError(Exception):
+    pass
+
+
+def check_json_serializable(obj: Any) -> None:
+    try:
+        _ = json.dumps(obj)
+    except Exception as e:
+        raise JSONSerializationError from e
+
+
+def is_json_serializable(obj: Any) -> bool:
+    try:
+        check_json_serializable(obj)
+        return True
+    except JSONSerializationError:
+        return False
+
+
+def dump_object(obj: Any) -> Any:
+    """Convert an object to a JSON-serializable object if it is an instance of `dataclasses.dataclass` or `pydantic.BaseModel`."""
+    if dataclasses.is_dataclass(obj) and not isinstance(obj, type):
+        return dataclasses.asdict(obj)
+    if isinstance(obj, BaseModel):
+        return obj.model_dump()
```

Comment:
I know that they changed this b/w version 1 and 2. Many environments still use pydantic 1, which has a `dict()` method.

We can potentially just avoid the pydantic stuff altogether and just check if it has a `model_dump()` and then a `dict()` method?